torchvision/csrc/io/image/cuda/readjpeg_cuda.cpp

```diff
@@ -0,0 +1,158 @@
+#include "readjpeg_cuda.h"
+
+#include <string>
+
+#if !NVJPEG_FOUND
+
+torch::Tensor decodeJPEG_cuda(const torch::Tensor& data, ImageReadMode mode) {
+  TORCH_CHECK(
+      false, "decodeJPEG_cuda: torchvision not compiled with nvJPEG support");
+}
+
+#else
+
+#include <ATen/ATen.h>
+#include <ATen/cuda/CUDAContext.h>
+#include <nvjpeg.h>
+
+static nvjpegHandle_t nvjpeg_handle = nullptr;
```

Comment:
Is there device-specific state associated with the nvjpegHandle? i.e. is it safe/optimal to create a nvjpeg_handle on one device, then switch CUDA devices and use the previously created nvjpeg_handle?

Reply:
It isn't completely obvious...although these parts of DALI give some clues

https://github.com/NVIDIA/DALI/blob/25ef92a7f9d48f1c895dcd7cceb9d98655aae279/dali/operators/decoder/nvjpeg/nvjpeg_decoder_gpu.h#L36-L37

https://github.com/NVIDIA/DALI/blob/25ef92a7f9d48f1c895dcd7cceb9d98655aae279/dali/operators/decoder/nvjpeg/nvjpeg_decoder_gpu.h#L166-L172

Reply:
The [docs](https://docs.nvidia.com/cuda/archive/11.2.2/nvjpeg/index.html#nvjpeg-opaque-lib-handle-struct) say:

> The library handle is used in any consecutive nvJPEG library calls, and should be initialized first. The library handle is thread safe, and can be used by multiple threads simultaneously.

With the links above where the struct is just defined as a global variable, and the fact that they call it a "library handle", I'd say it's fair to assume that it's not device-specific.

From a quick test with 2 GPUs, this passes fine:

```py
                img_nvjpeg = decode_jpeg(data, mode=mode, device='cuda:0')
                img_nvjpeg2 = decode_jpeg(data, mode=mode, device='cuda:1')
                self.assertTrue((img_nvjpeg.cpu().float() - img_nvjpeg2.cpu().float()).abs().mean() < 1e-10)
 ```